vpr/src/analytical_place/ap_flow_enums.h

```diff
@@ -0,0 +1,20 @@
+/**
+ * @file
+ * @author  Alex Singer
+ * @date    February 2025
+ * @brief   Enumerations used by the Analytical Placement Flow.
+ */
+
+#pragma once
+
+/**
+ * @brief The type of a Full Legalizer.
+ *
+ * The Analytical Placement flow may implement different Full Legalizers. This
+ * enum can select between these different Full Legalizers.
+ */
+enum class e_ap_full_legalizer {
+    Naive,  ///< The Naive Full Legalizer, which builds and places cluster in the first possible place.
```

Comment:
Expanding on this would be useful ... what does 'first possible place' mean?  Does it pay attention to a flat placement at all?